src/schema/slack/functions/_scripts/src/templates/test_template.ts

```diff
@@ -0,0 +1,137 @@
+import {
+  autogeneratedComment,
+  getFunctionName,
+  getSlackCallbackId,
+  renderFunctionImport,
+  renderTypeImports,
+} from "./template_utils.ts";
+import { FunctionParameter, FunctionRecord } from "../types.ts";
+import { manifestFunctionFieldsToTypeScript } from "./template_function.ts";
+import SchemaTypes from "../../../../../schema_types.ts";
+
+export const manifestFunctionToTypeScript = (
+  functionRecord: FunctionRecord,
+) => {
+  return `{${manifestFunctionFieldsToTypeScript(functionRecord)}}`;
+};
+
+const renderFunctionManifestTest = (
+  functionRecord: FunctionRecord,
+) => {
+  const functionName = getFunctionName(functionRecord.callback_id);
+  const typescript: string[] = [];
+  typescript.push(
+    `assertEquals(${functionName}.definition.callback_id, "${
+      getSlackCallbackId(functionRecord)
+    }");`,
+  );
+  typescript.push(
+    `const expected: ManifestFunctionSchema = ${
+      manifestFunctionToTypeScript(
+        functionRecord,
+      )
+    };`,
+  );
+  typescript.push(`const actual = ${functionName}.export();`);
+  typescript.push("");
+  typescript.push(`assertEquals(actual, expected);`);
+  return `() => {${typescript.join("\n")}}`;
+};
+
+const workflowToTypeScript = (functionName: string) => {
+  const typescript: string[] = [];
+  typescript.push(`callback_id: "test_${functionName}_built_in"`);
+  typescript.push(`title: "Test ${functionName}"`);
+  typescript.push(
+    `description: "This is a generated test to test ${functionName}"`,
+  );
+  return `{${typescript.join(", \n")}}`;
+};
+
+const getInputValue = (name: string, type: string) => {
+  switch (type) {
+    case SchemaTypes.integer:
+      return `${name}: 1`;
+    case SchemaTypes.boolean:
+      return `${name}: true`;
+    case SchemaTypes.number:
+      return `${name}: 1.5`;
+    case SchemaTypes.array:
+      return `${name}: []`;
+    case SchemaTypes.object:
+      return `${name}: {}`; // TODO make this recursive
+    default:
+      return `${name}: "test"`;
+  }
+};
```

Comment:
Honestly, our typing on step inputs is very liberal and we intentionally always need to support them accepting a string since step inputs can be a templated variable reference that looks something like `"{{inputs.my_channel}}"`

Because of this, we could probably just assign them all to a string and get rid of this switch case.